src/components/MessageInput/CooldownTimer.tsx

```diff
@@ -4,20 +4,24 @@ export type CooldownTimerProps = {
   cooldownInterval: number;
   setCooldownRemaining: React.Dispatch<React.SetStateAction<number | undefined>>;
 };
-export const CooldownTimer = ({ cooldownInterval, setCooldownRemaining }: CooldownTimerProps) => {
-  const [seconds, setSeconds] = useState(cooldownInterval);
+export const CooldownTimer = ({ cooldownInterval }: CooldownTimerProps) => {
+  const [seconds, setSeconds] = useState<number | undefined>();
 
   useEffect(() => {
-    const countdownInterval = setInterval(() => {
-      if (seconds > 0) {
+    let countdownTimeout: ReturnType<typeof setTimeout>;
+    if (seconds > 0) {
```

Comment:
```suggestion
    if (typeof seconds === 'number' && seconds > 0) {
```
Just to cover this error:
`'seconds' is possibly 'undefined'.`

Reply:
@arnautov-anton I think we do not have to check for undefined here as you can compare undefined with number:

![image](https://github.com/GetStream/stream-chat-react/assets/32706194/ce2ab763-75aa-4c0e-b4e9-b4a3c1fa6d9f)